client/my-sites/checkout/composite-checkout/hooks/wpcom-store.ts

```diff
@@ -188,17 +177,24 @@ export function useWpcomStore(): WpcomCheckoutStore | undefined {
 		}
 	}
 
-	return registerStore( STORE_KEY, {
+	return createReduxStore( CHECKOUT_STORE_KEY, {
 		reducer( state: WpcomStoreState | undefined, action: WpcomStoreAction ): WpcomStoreState {
-			const checkedState =
-				state === undefined ? getInitialWpcomStoreState( emptyManagedContactDetails ) : state;
+			if ( action.type === 'RESET' || state === undefined ) {
+				state = getInitialWpcomStoreState( emptyManagedContactDetails );
+			}
 			return {
-				contactDetails: contactReducer( checkedState.contactDetails, action ),
-				recaptchaClientId: recaptchaClientIdReducer( checkedState.recaptchaClientId, action ),
-				vatDetails: vatDetailsReducer( checkedState.vatDetails, action ),
+				contactDetails: contactReducer( state.contactDetails, action ),
+				recaptchaClientId: recaptchaClientIdReducer( state.recaptchaClientId, action ),
+				vatDetails: vatDetailsReducer( state.vatDetails, action ),
 			};
 		},
 		actions,
 		selectors,
 	} );
 }
+
+const store = createStore();
+
+register( store );
+
+export const CHECKOUT_STORE = store;
```

Comment:
That's a nice way to enforce the store export to have a particular unique name, cc @noahtallen

Reply:
Thanks. I thought I copied this pattern from another one of the diffs off of https://github.com/Automattic/wp-calypso/issues/74399 but... now I'm not sure if maybe I invented it accidentally?

Reply:
True, we could even still use `export const CHECKOUT_STORE = createReduxStore( ... )` etc!

I wonder if we should decide on a naming pattern to use. I've been using a format like `import { store as checkoutStore }`

Reply:
I think we should go with checkoutStore, since All_CAPS style is for constants, but this is no longer a constant.